R/utils.R

```diff
@@ -259,82 +259,500 @@ rowcol_to_dim <- function(row, col) {
+#'    it improves the selection of various parts of `x`
+#'    One of "x", "data", "col_names", or "row_names".
+#'    `"data"` will only select the data part, excluding row names and column names  (default if `cols` or `rows` are specified)
+#'    `"x"` Includes the data, column and row names if they are present. (default if none of `rows` and `cols` are provided)
+#'    `"col_names"` will only return column names
+#'    `"row_names"` Will only return row names.
+#'
+#' @return A `dims` string
+#' @export
 #' @examples
-#' # either vectors
+#' # Provide coordinates
+#' wb_dims()
+#' wb_dims(1, 4)
+#' wb_dims(rows = 1, cols = 4)
+#' wb_dims(from_row = 4)
+#' wb_dims(from_col = 2)
+#' wb_dims(from_col = "B")
+#' wb_dims(1:4, 6:9, from_row = 5)
+#' # Provide vectors
+#' wb_dims(1:10, c("A", "B", "C"))
 #' wb_dims(rows = 1:10, cols = 1:10)
+#' # provide `from_col` / `from_row`
+#' wb_dims(rows = 1:10, cols = c("A", "B", "C"), from_row = 2)
+#' wb_dims(rows = 1:10, cols = 1:10, from_col = 2)
+#'
 #' # or objects
-#' wb_dims(mtcars)
-#' @export
-wb_dims <- function(...) {
-
+#' wb_dims(x = mtcars, col_names = TRUE)
+#' # select all data
+#' wb_dims(x = mtcars, select = "data")
+#' # column names of an object (with the special select = "col_names")
+#' wb_dims(x = mtcars, select = "col_names")
+#' # usually, it's better
+#' # dims of all the data of mtcars. (when not using name)
+#' wb_dims(x = mtcars, col_names = FALSE)
+#'
+#' # dims of the column names of an object
+#' wb_dims(x = mtcars, select = "col_names", col_names = TRUE)
+#'
+#' ## add formatting to column names with the help of `wb_dims()`----
+#' wb <- wb_workbook()
+#' wb$add_worksheet("test wb_dims() with an object")
+#' dims_mtcars_and_col_names <- wb_dims(x = mtcars)
+#' wb$add_data(x = mtcars, dims = dims_mtcars_and_col_names)
+#'
+#' # Put the font as Arial for the data
+#' dims_mtcars_data <- wb_dims(x = mtcars, select = "data")
+#' wb$add_font(dims = dims_mtcars_data, name = "Arial")
+#'
+#' # Style col names as bold using the special `select = "col_names"` with `x` provided.
+#' dims_column_names <- wb_dims(x = mtcars, select = "col_names")
+#' wb$add_font(dims = dims_column_names, bold = TRUE, size = 13)
+#'
+#' # Finally, to add styling to column "cyl" (the 4th column) (only the data)
+#' # there are many options, but here is the preferred one
+#' # if you know the column index, wb_dims(x = mtcars, cols = 4) also works.
+#' dims_cyl <- wb_dims(x = mtcars, cols = "cyl")
+#' wb$add_fill(dims = dims_cyl, color = wb_color("pink"))
+#'
+#' # Mark a full column as important(with the column name too)
+#' wb_dims_vs <- wb_dims(x = mtcars, cols = "vs", select = "x")
+#' wb$add_fill(dims = wb_dims_vs, fill = wb_color("yellow"))
+#' wb$add_conditional_formatting(dims = wb_dims(x = mtcars, cols = "mpg"), type = "dataBar")
+#' # wb_open(wb)
+wb_dims <- function(..., select = NULL) {
   args <- list(...)
-  nams <- names(args)
+  lengt <- length(args)
```

Comment:
I'd prefer `len`, but ignore that 😄

Reply:
I changed it ;), just replace all lengt -> len, then  replace all lenh -> length ;) 2 seconds